unity/Assets/Maroon/reusableScripts/Physics/Electromagnetism/IField.cs

```diff
@@ -83,4 +83,8 @@ public abstract class IField : MonoBehaviour
     public abstract float getStrength(Vector3 position);
 
     public abstract float getStrengthInPercent(Vector3 position);
+
+    public abstract void addProducerToSet(GameObject producer);
+
+    public abstract void removeProducerFromSet(GameObject producer);
```

Comment:
I am not an expert in all this logic, so please correct me if I'm wrong, but Is it necessary to add these to the abstract IField? Because if BField does not support these and throws an exception, maybe adding these to the EField method only would make more sense?